src/views/bouquets/BouquetEditView.vue

```diff
@@ -28,43 +50,73 @@ const search = async (query) => {
   }).filter(r => !datasets.value.map(d => d.id).includes(r.value))
 }
 
-const onSelect = async (value) => {
-  const dataset = await datasetStore.load(value)
-  if (dataset) {
-    datasets.value.push(dataset)
+const onSubmitModal = async () => {
+  if (!selectedDataset.value.id) return
+  if (isEditDesc.value) {
+    const idx = datasets.value.findIndex(d => d.dataset.id === selectedDataset.value.id)
+    datasets.value[idx].description = selectedDataset.value.description
+  } else {
+    const dataset = await datasetStore.load(selectedDataset.value.id)
+    if (dataset) {
+      datasets.value.push({ dataset, description: selectedDataset.value.description })
+    }
   }
-  selector.value.clear()
-  selector.value.refreshOptions()
+  selectedDataset.value = {}
+  isEditDesc.value = false
 }
 
 const onDeleteDataset = (datasetId) => {
-  datasets.value = datasets.value.filter(d => d.id !== datasetId)
+  datasets.value = datasets.value.filter(d => d.dataset.id !== datasetId)
+  delete loadedBouquet.value.extras[`ecospheres:${datasetId}:description`]
+}
+
+const onEditDataset = (dataset) => {
+  isModalOpen.value = true
+  isEditDesc.value = true
+  selectedDataset.value.id = dataset.id
+  selectedDataset.value.title = dataset.title
+  selectedDataset.value.description = datasets.value.find(d => d.dataset.id === dataset.id).description
+}
+
+const extrasFromDatasets = () => {
+  const extras = {}
+  for (const dataset of datasets.value) {
+    if (dataset.description) {
```

Comment:
Fails to cover case where description exists and is removed during edit.

Reply:
True. I fixed in the simplest way possible, but now we're storing empty values for each and every dataset even if no description is set. Not a big problem at this stage I guess, and it avoids complexity when saving the extras while preserving the previous ones. This could be better if we have a list or a dict dedicated to Ecospheres in extras.